dags/automated_transformation/transformation_functions.py

```diff
@@ -0,0 +1,207 @@
+import re
+from datetime import datetime
+
+import boto3
+import xarray
+
+
+def get_all_s3_keys(bucket, model_name, ext):
+    """Function fetches all the s3 keys from the given bucket and model name.
+
+    Args:
+        bucket (str): Name of the bucket from where we want to fetch the data
+        model_name (str): Dataset name/folder name where the data is stored
+        ext (str): extension of the file that is to be fetched.
+
+    Returns:
+        list : List of all the keys that match the given criteria
+    """
+    session = boto3.session.Session()
+    s3_client = session.client("s3")
+    keys = []
+
+    kwargs = {"Bucket": bucket, "Prefix": f"{model_name}"}
+    while True:
+        resp = s3_client.list_objects_v2(**kwargs)
+        print("response is ", resp)
+        for obj in resp["Contents"]:
+            if obj["Key"].endswith(ext) and "historical" not in obj["Key"]:
+                keys.append(obj["Key"])
+
+        try:
+            kwargs["ContinuationToken"] = resp["NextContinuationToken"]
+        except KeyError:
+            break
+
+    print(f"Discovered {len(keys)}")
+    return keys
+
+
+"""
+The naming convention for the transformation function is as follows:
+collectionname_transformation
+where
+1. collection name refers to the STAC collection name that we want to provide
+   for the given dataset. Make sure that the collection name in the function
+   is same to the one passed as argument when running the DAG.
+2. Collection name will be followed by the word transformation which will
+   differentiate the transformation functions from any other functions in
+   file.
+"""
+
+
+def tm54dvar_ch4flux_mask_monthgrid_v5_transformation(file_obj, name, nodata):
```

Comment:
i'm not sure about this. @amarouane-ABDELHAK let's discuss. we can't be putting ghg-specific things in generic veda repositories and vice versa.

Reply:
Make sense, I think these particular file can be in S3 (GHG S3 bucket)